src/main/kotlin/lotto/model/winning/WinningPlace.kt

```diff
@@ -0,0 +1,34 @@
+package lotto.model.winning
+
+import lotto.model.Money
+
+enum class WinningPlace(val price: Money, val count: Int, val bonusExist: Boolean) {
+    FIRST(Money(2_000_000_000), 6, false),
+    SECOND(Money(30_000_000), 5, true),
+    THIRD(Money(1_500_000), 5, false),
+    FOURTH(Money(50_000), 4, false),
+    FIFTH(Money(5_000), 3, false),
+    MISS(Money(0), 0, false);
+
+    companion object {
+        fun match(winningCount: Int, bonusExist: Boolean): WinningPlace {
+            val candidates = values().filter {
+                it.count == winningCount
+            }
+
+            if (isSecondOrThird(candidates)) {
+                return candidates.last { it.bonusExist == bonusExist }
+            }
+
+            if (candidates.isEmpty()) {
+                return MISS
+            }
+
+            return candidates.first()
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
            return candidates.firstOrNull() ?: MISS
```